includes/site-health/audit-autoloaded-options/hooks.php

```diff
@@ -26,3 +26,86 @@ function perflab_aao_add_autoloaded_options_test( $tests ) {
 	return $tests;
 }
 add_filter( 'site_status_tests', 'perflab_aao_add_autoloaded_options_test' );
+
+/**
+ * Register admin actions for handling autoload enable/disable.
+ *
+ * @since n.e.x.t
+ */
+function perflab_aao_register_admin_actions() {
+	add_action( 'admin_action_perflab_aao_update_autoload', 'perflab_aao_handle_update_autoload' );
+}
+add_action( 'admin_init', 'perflab_aao_register_admin_actions' );
+
+/**
+ * Callback for handling disable autoload action.
+ *
+ * @since n.e.x.t
+ */
+function perflab_aao_handle_update_autoload() {
+	check_admin_referer( 'perflab_aao_update_autoload' );
+
+	if ( ! isset( $_GET['option_name'], $_GET['autoload'] ) ) {
+		wp_die( esc_html__( 'Missing required parameter.', 'performance-lab' ) );
+	}
+
+	$option_name  = sanitize_text_field( wp_unslash( $_GET['option_name'] ) );
+	$autoload     = isset( $_GET['autoload'] ) ? rest_sanitize_boolean( $_GET['autoload'] ) : false;
+	$value_length = isset( $_GET['value_length'] ) ? (int) $_GET['value_length'] : 0;
+
+	if ( ! current_user_can( 'manage_options' ) ) {
+		wp_die( esc_html__( 'Permission denied.', 'performance-lab' ) );
+	}
+
+	if ( empty( $option_name ) ) {
+		wp_die( esc_html__( 'Invalid option name.', 'performance-lab' ) );
+	}
+
+	// Check if the option exists.
+	if ( false === get_option( $option_name ) ) {
+		// Option doesn't exist, return an error or handle the situation accordingly.
+		wp_die( esc_html__( 'The option does not exist.', 'performance-lab' ) );
+	}
+
+	$result = wp_set_option_autoload( $option_name, $autoload );
+
+	if ( $result ) {
+		// Update modified options list.
+		$modified_options = get_option( 'perflab_aao_modified_options', array() );
+		if ( ! $autoload ) {
+			$modified_options[ $option_name ] = $value_length;
+		} elseif ( isset( $modified_options[ $option_name ] ) ) {
+			unset( $modified_options[ $option_name ] );
+		}
+		update_option( 'perflab_aao_modified_options', $modified_options );
```

Comment:
Maybe better to call this `disabled_options` rather than `modified_options`? This better indicates that autoload for those options was disabled, rather than just "modified", which could be in any way.